### DISCUSSION TITLE
[Feature Suggestion] Ability to run a test from cli without overriding parameters with TestContext and without using TestCaseSource object

### DISCUSSION THREAD

**Discussion by arielnahom** [nunit/nunit-console#1527] [Ideas]
Hi, I'd like to have the ability to run a specific test (already possible with the cli) with specific parameters seamlessly (and integrate it with other DevOps tools to run automatically).

currently if i want to do it - I need to override each param on each method with TestContext.Parameters.Get which is not ideal and error prone.
if nunit will just populate the parameters of the function which has same name it would be much more elegant.

Also - if I run it like so 
`nunit3-console .\bin\local\Debug\net48\TestProject1.dll --where "cat == A.C" -tp "ProductId=6"`
and it will run 3 times - the same test. if I hade 40 cases in ProductDiscountsTestsCases as TestCaseSource it will run 40 times *the same test*.

I'd like to be able to override and ignore the TestCaseSource when I run from the nunit3-console cli and use params provided in cli and run it only once.
It enables much more integration with automatic tools and other CI/CD tools to automate things with greater flexibility.
Thanks.

```csharp
        public class ProductDiscountsTests
        {
            [TestCaseSource(typeof(ProductDiscountsTestsCases), nameof(ProductDiscountsTestsCases.DiscountCases), Category = "A.C")]
            public void DivideTest(Product product)
            {
                product.ProductId = TestContext.Parameters.Get("ProductId", product.ProductId);
                Assert.True(product.ProductId > 0);
            }
        }

        public class ProductDiscountsTestsCases
        {
            public static IEnumerable DiscountCases
            {
                get
                {
                    yield return new TestCaseData(new Product { ProductId = 1 });
                    yield return new TestCaseData(new Product { ProductId = 1 });
                    yield return new TestCaseData(new Product { ProductId = 3 });
                }
            }

        }
        public class Product
        {
            public int ProductId { get; set; }
        }
```

**Comment by CharliePoole** [comment]
As you suggest, that would be a new feature. I just wanted to mention that `TestContext.Parameters` is in no way intended to deal with individual method parameters, but with overall parameters for the run itself.

Currently, you can select individual tests (i.e. one of your 40 sets of parameters) only if you can figure out what name NUnit has given to that test. That's not difficult for simple type arguemts like int or string but becomes more complicated when complex types are involved. There is no way to run a particular method with a new set of arguments, not already specified in a test case source.
